content/My favorite extensions for programming Ruby in Visual Studio Code.md

```diff
@@ -0,0 +1,106 @@
+
+- Context-aware suggestions and documentation for the Ruby core
+- Detection of some variable types and method return values (e.g., String.new. returns String instance methods)
+- Identification of local, class, and instance variables within the current scope
+- Find references and rename symbols (experimental as of solargraph gem v0.22.0)
+- Support for gems
+- Near-complete support for the Ruby core and stdlib
+- Partial support for Ruby on Rails
+
+![Ruby Solargraph](images/rubysolargraph.gif)
+
+## [ruby-rubocop](https://marketplace.visualstudio.com/items?itemName=misogi.ruby-rubocop)
+
+Having rubocop configured in our ruby project will help us to maintain the quality of the code, especially if our project is being developed by a team. This extension will let us autoformat and correct our code basing it the directives written in the .rubucop.yml file. With only a key shortcut your code will be formatted in a second, is fast, is good, is a **must-have** extension.
+
+**Features:**
+
+- Lint by executing the command "Ruby: lint by rubocop" (cmd+shift+p and type command)
+- Auto invoke when saving file
+- Auto correct command "Ruby: autocorrect by rubocop"
+
+![Ruby Solargraph](images/rubyrubocop.gif)
+
+## [Prettier](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode)
+
+This extension will help us to format the code of Javascript, HTML, CSS, JSON, and many more files. Is simple and quick, but is not so much configurable like other code-formatter extensions.
+
+**Features:**
+
+- Supports many languages
+- You press save and code is formatted
+- No need to discuss style in code review
+- Saves you time and energy
+
+## [VSCode Great Icons](https://marketplace.visualstudio.com/items?itemName=emmanuelbeziat.vscode-great-icons)
+
+A very simple one, adds icons to our files, it personally helps me a lot to know instantly which kind of file I'm seeing.
+
+![VSCode Great Icons](images/vscodegreaticons.jpg)
+
+
+# I&#39;ve been using atom and now I switched to Visual Studio Code, I&#39;m a bit lost
+
+I fully understand you, I miss so much the Atom shortcut keymap when I made the switch, but here is your salvation.
+
+## [Atom Keymap](https://marketplace.visualstudio.com/items?itemName=ms-vscode.atom-keybindings)
+
+This extension ports popular Atom keyboard shortcuts to Visual Studio Code helps you a lot to make easier the switch between the code editors.
+
+## [Atom One Dark Theme](https://marketplace.visualstudio.com/items?itemName=akamud.vscode-theme-onedark)
+
+And the last extension that I will share with you is an aesthetic one, when I started using Visual Studio Code I missed the theme that Atom has, so this extension fix that problem for me.
+
+# Conclusion
+
+Those are my favorite extension for Visual Studio Code at the moment that I'm writing this, I hope that they help you to improve your experience using this famous code editor, comment below this article if you have any other extensions recommendation!
```

Comment:
I like the part about the comments, it helps with readers engagement 👏

Reply:
yes!! I thought the same!